### PR TITLE
[ryml] [c4core] Update RapidYaml (ryml) to newer version

### LINKED ISSUE #17972: [ryml] update to latest master
Library name: ryml

New version number: move from previous head of master to current head of master.

Other information that may be useful (release notes, etc...)

### PR DESCRIPTION
**Pull request by MichaelGoulding**
Fixes #17972

### CODE REVIEW

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]
@MichaelGoulding ,we should replace "
````
file(INSTALL    "${SOURCE_PATH}/LICENSE.txt"    DESTINATION "${CURRENT_PACKAGES_DIR}/share/c4core" RENAME copyright)
````
to"
````
file(INSTALL    "${SOURCE_PATH}/LICENSE.txt"    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
````
"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]
We should replace "
````
file(INSTALL    "${SOURCE_PATH}/LICENSE.txt"    DESTINATION "${CURRENT_PACKAGES_DIR}/share/ryml" RENAME copyright)
" 
````
to "
````
file(INSTALL    "${SOURCE_PATH}/LICENSE.txt"    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
````
"

**Review by JonLiu1993 — APPROVED** [review]

File: ports/ryml/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "ryml",
+  "version-string": "2021-05-27",
```
Comment (JonLiu1993) [inline ports/ryml/vcpkg.json:3]: @MichaelGoulding ,We should change "version-string" to "version-date"

File: versions/r-/ryml.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "a1732ca3b070c5e730ec67d904246946b9b9aa01",
+      "version-string": "2021-05-27",
```
Comment (JonLiu1993) [inline versions/r-/ryml.json:5]: We should change "version-string" to "version-date"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/ryml/cmake-fix.patch [hunk 3]
```diff
@@ -2,12 +2,6 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
 index 3e4eb23..115b8aa 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -1,4 +1,5 @@
- cmake_minimum_required(VERSION 3.9)
-+
- project(ryml)
- 
- include(./ext/c4core/cmake/c4Project.cmake)
```
Comment (JonLiu1993) [inline ports/ryml/cmake-fix.patch:10]: @MichaelGoulding ,Could you please clarify why these three lines are deleted

File: versions/c-/c4core.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "12b84a31469a78dd4b42dcf58a27d4600f6b2d48",
+      "version-date": "2021-01-14",
+      "port-version": 0
+    },
+    {
+      "git-tree": "66c4ceba9d7d73a194de651f9e8ca7b87f595e23",
+      "version-date": "2021-01-14",
+      "port-version": 1
+    },
```
Comment (JonLiu1993) [inline versions/c-/c4core.json:12]: We only add ‘version’ once for each update

File: versions/r-/ryml.json [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "783be628f42faca427a8bc1394ccd20580ef56c8",
+      "version-date": "2021-05-27",
+      "port-version": 0
+    },
+    {
+      "git-tree": "4d15a95ca4e2cb0b2c4d3adc7a8e5c55da878ff9",
+      "version-date": "2021-05-27",
+      "port-version": 1
+    },
```
Comment (JonLiu1993) [inline versions/r-/ryml.json:12]: We only add ‘version’ once for each update, so please delete these lines

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: versions/r-/ryml.json (hunk 5 above)
Comment (MichaelGoulding) [reply versions/r-/ryml.json]: Done.  @JonLiu1993 , please remove.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the updates!

**Pull request merged**